edx_exams/apps/core/models.py

```diff
@@ -241,11 +241,20 @@ def get_attempt_for_user_with_attempt_number_and_resource_id(cls, user_id, attem
         attempt number described by attempt_number.
         """
         try:
-            attempt = ExamAttempt.objects.get(
-                user_id=user_id,
-                attempt_number=attempt_number,
-                exam__resource_id=resource_id
-            )
+            # If user_id is an integer, use our internal database ID
```

Comment:
I'm wondering if we should split this logic between two different model functions, with the if/else statement happening in api.py instead.

Reply:
tbh I wouldn't hide this behavior all the way down here. You can simply get the User in the view `user = User.object.get(anonymous_user_id=<sub_id>)` and than pass user.id down into the existing functions we have. More code reuse and we don't have multiple concepts of what user_id means.

The extra DB call is pretty much irrelevant performance wise. We don't need to over-optimize such that everything is one ORM call.